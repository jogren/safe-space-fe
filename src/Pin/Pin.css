.Pin_img {
  height: 20px;
  width: 20px;
  transition: all .2s ease-in-out;
}

.pin-hover {
  padding: 10px;
  background-color: black;
  opacity: 0.8;
  position: absolute;
  z-index: 99;  
  width: 200px;
}

.pin-hover p,
.pin-hover h3 {
  color: white;
}
