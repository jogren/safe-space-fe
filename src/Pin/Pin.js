import React, { useContext } from 'react';
import { CurrentCenterContext } from '../CurrentCenterContext';
import './Pin.css';

const Pin = ({ selectPin, center }) => {
  const { reliefCenter, setreliefCenter } = useContext(CurrentCenterContext);


  return(
    <img onClick={() => setreliefCenter(center)} className="Pin_img" src={'https://cdn4.iconfinder.com/data/icons/pictype-free-vector-icons/16/location-alt-512.png'} alt="hello"/>
  )
}

export default Pin;