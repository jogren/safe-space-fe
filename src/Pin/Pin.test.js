import React from 'react';
import { shallow } from 'enzyme';
import { Pin } from './Pin';


describe('Pin', () => {
  let wrapper;
  let mockSelectCenter = jest.fn();
  let mockCenter = { 
    name: "Central Denver Disaster Relief", 
    addressPrint: "601 W 11th Ave Denver CO 80204", 
    lat: "39.7341653", 
    lng: "-104.9974374", 
    phone: "5555555555",
    website: "www.safe-space.com",
    id: "2"
  }
  let mockReliefCenter = {
    addressPrint: "1331 17th St Denver CO 80202",
    email: "get-relief@this-center.com",
    id: 1,
    lat: "39.7508132",
    lng: "-104.9967997",
    name: "Turing Relief Center",
    phone: "5555555555",
    website: "www.relief-center.com"
  }

  beforeEach(() => {
    wrapper = shallow(<Pin 
      center={mockCenter}
      selectCenter={mockSelectCenter}
      reliefCenter={mockReliefCenter}
    />)
  })

  it.skip('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})