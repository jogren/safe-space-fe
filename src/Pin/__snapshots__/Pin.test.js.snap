// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin should match the snapshot with the data passed through 1`] = `undefined`;
